pdr_backend/lake/table_silver_pdr_predictions.py

```diff
@@ -0,0 +1,283 @@
+from typing import Dict, Tuple
+
+import polars as pl
+from enforce_typing import enforce_types
+from polars import Boolean, Float64, Int64, Utf8, DataFrame
+from pdr_backend.lake.table import Table
+from pdr_backend.ppss.ppss import PPSS
+from pdr_backend.util.constants_opf_addrs import get_opf_addresses
+
+
+silver_pdr_predictions_table_name = "silver_pdr_predictions"
+
+# JOIN DATA TO ENRICHED PREDICTOOR PREDICTIONS SCHEMA
+silver_pdr_predictions_schema = {
+    "ID": Utf8,  # f"{contract}-{slot}-{user}"
+    "slot_id": Utf8,  # f"{contract}-{slot}"
+    "contract": Utf8,  # f"{contract}"
+    "slot": Int64,
+    "user": Utf8,
+    "pair": Utf8,
+    "timeframe": Utf8,
+    "source": Utf8,
+    "predvalue": Boolean,
+    "truevalue": Boolean,
+    "stake": Float64,
+    "payout": Float64,
+    "sum_stake": Float64,
+    "sum_revenue": Float64,
+    "sum_revenue_df": Float64,
+    "sum_revenue_user": Float64,
+    "sum_revenue_stake": Float64,
+    "win": Boolean,
+    "count_wins": Int64,
+    "count_losses": Int64,
+    "count_predictions": Int64,
+    "timestamp": Int64,
+    "last_event_timestamp": Int64,
+}
+
+timeframe_to_seconds = {
+    "5m": 300,
+    "1h": 3600
+}
+
+def update_fields(
+    current_df: dict,
+    previous_df: DataFrame,
+    df_subscriptions: DataFrame,
+    user_subscriptions: DataFrame,
+    df_slots: DataFrame,
+) -> dict:
```

Comment:
This looks great @KatunaNorbert!

It provides an outline for all the calculations we'll have to do for the silver table, for each individual parameter... awesome! Now that we know these, we'll be able to more easily write them in SQL.
1. Don't worry about cleaning this up, splitting into functions, etc... 
2. Focus on improving test coverage such that you can validate that the values are calculated correctly
3. Once you feel tests are validated well enough, let's move onto streamlit.

To move onto streamlit:
1. Freeze this PR (stop changing it)
2. Look at Calina's PR for streamlit, and replicate the cli + streamlit_entrypoint, such that it's easy for you to get the same streamlit environment going
3. Look at Calina's changes for how to setup the sim_plotter page, how to setup columns, contianers, and how to initialize plots
4. Create your own PredictoorIncomePlotter class, and implement plots that use data from silver_pdr_predictions table to visualize all data.

I'll update the analytics/dashboard epic with this info.